<a href="https://colab.research.google.com/github/AEIBIANCOFCE/AnalisisEstadisticoI/blob/main/5%20Test%20de%20Hip%C3%B3tesis/7_AEI_Test_de_Hip%C3%B3tesis_Python.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<figure>
<center>
<img src='https://www.economicas.uba.ar/wp-content/uploads/2020/08/cropped-logo_FCE.png' />
</figure>

# **Universidad de Buenos Aires**
## **Facultad de Ciencias Económicas**

### **Análisis Estadístico I**

### Cátedra: Bianco

#### **Test de Hipótesis con Python**

***Material elaborado por la profesora Natalia Salaberry***

###**Librerías**

In [ ]:
import scipy.stats as ss
import numpy as np

## **Para una población**

####**TH para la media con varianza poblacional conocida**



Sea $X \sim N(\mu ; \sigma )$

${Z_{obs} =} {{\bar{X} - \mu } \over {\sigma \over \sqrt n}}$ $\sim N(0 ; 1)$

Hipótesis y Criterio de decisión (C.D) para rechazar H0

Prueba unilateral a derecha
\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: \mu \leq a     \\
		 H1: \mu > a   
	       \end{array}
 \thinspace C.D: si \thinspace Z_{obs} > Z_{1-\alpha} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

Prueba unilateral a izquierda

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: \mu \geq a    \\
		 H1: \mu < a        
	       \end{array}
\thinspace C.D: si \thinspace Z_{obs} < Z_{\alpha} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

Prueba bilateral

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: \mu = a     \\
		 H1: \mu \neq a        
	       \end{array}
	\thinspace C.D: si \thinspace |Z_{obs}| > Z_{1-{\alpha \over 2}} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

Ejemplo: Sea X con distribución normal con media 14 y un desvío estándar de 0.8. Se toma una muestra de 16 y se obtuvo una media de 13.5. Se quiere decidir si la media poblacional es menor a 14 con α = 0.05

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: \mu \geq 14    \\
		 H1: \mu < 14        
	       \end{array}
\thinspace C.D: si \thinspace Z_{obs} < Z_{\alpha} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

In [ ]:
media_muestral= 13.5
mu= 14
sigma= 0.8
n= 16
alfa= 0.05
simbolo_H1='<' #las opciones de símbolo son '<' '>' ó '' para distinto


Zobs=round((media_muestral-mu) / (sigma/np.sqrt(n)),4)
if simbolo_H1=='>':
  Zalfa  = round(ss.norm.ppf(1-alfa, 0,1),4)
  if Zobs > Zalfa:
    respuesta='Como se cumple C.D ya que ' + 'Zobs='+str(Zobs)+' es mayor que '+ 'Z_1-alfa='+str(Zalfa)+' entonces se rechaza H0'
  else:
    respuesta='Como no se cumple C.D ya que ' + 'Zobs='+str(Zobs)+' es menor que '+ 'Z_1-alfa='+str(Zalfa)+' entonces no se rechaza H0'

    p_valor=1-ss.norm.cdf(Zobs, 0,1)
    if p_valor < alfa:
      respuesta1='Como se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es menor que '+ 'alfa='+ str(alfa)+' entonces se rechaza H0'
    else:
      respuesta1='Como no se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es mayor que '+ 'alfa='+ str(alfa)+' entonces no se rechaza H0'

elif simbolo_H1=='<':
  Zalfa  = round(ss.norm.ppf(alfa, 0,1),4)
  if Zobs < Zalfa:
    respuesta='Como se cumple C.D ya que ' + 'Zobs='+str(Zobs)+' es menor que '+ 'Zalfa='+str(Zalfa)+' entonces se rechaza H0'
  else:
    respuesta='Como no se cumple C.D ya que ' + 'Zobs='+str(Zobs)+' es mayor que '+ 'Zalfa='+str(Zalfa)+' entonces no se rechaza H0'

  p_valor=ss.norm.cdf(Zobs, 0,1)
  if p_valor < alfa:
    respuesta1='Como se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es menor que '+ 'alfa='+ str(alfa)+' entonces se rechaza H0'
  else:
    respuesta1='Como no se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es mayor que '+ 'alfa='+ str(alfa)+' entonces no se rechaza H0'

else:
  Zalfa  = round(ss.norm.ppf(1-alfa/2, 0,1),4)
  if np.abs(Zobs) > Zalfa:
    respuesta='Como se cumple C.D ya que ' + '|Zobs|='+'|'+str(Zobs)+'|'+' es mayor que '+ 'Z_1-alfa/2='+str(Zalfa)+' entonces se rechaza H0'
  else:
    respuesta='Como no se cumple C.D ya que ' + '|Zobs|='+'|'+str(Zobs)+'|'+' es menor que '+ 'Z_1-alfa/2='+str(Zalfa)+' entonces no se rechaza H0'

  p_valor=2*(1-ss.norm.cdf(abs(Zobs), 0,1))
  if p_valor < alfa:
    respuesta1='Como se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es menor que '+ 'alfa='+ str(alfa)+' entonces se rechaza H0'
  else:
    respuesta1='Como no se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es mayor que '+ 'alfa='+ str(alfa)+' entonces no se rechaza H0'

respuesta, respuesta1

('Como se cumple C.D ya que Zobs=-2.5 es menor que Zalfa=-1.6449 entonces se rechaza H0',
 'Como se cumple C.D ya que P-valor=0.006209665325776132 es menor que alfa=0.05 entonces se rechaza H0')

Se verifica este resultado del siguiente modo:

Dados $\bar{X}=13.5$ $\sigma=0.8$ $N=16$ y $\alpha=5\%$

Luego:

$Z_{\alpha}=Z_{0.05}= - 1.65$


${Z_{obs} =} {{\bar{X} - \mu } \over {\sigma \over \sqrt n}}={{13.5 - 14 } \over {0.8 \over \sqrt 16}}={-2.5}$

Como ${Z_{obs} =}{-2.5}$ es menor que $Z_{0.05}= - 1.65$ se cumple el criterio de decisión y por lo tanto se rechaza H0

Como $p_{valor}= p(Z < -2.5)=1-p(Z < 2.5)=1-0.9938=0.0062$ es menor que $\alpha = 0.05$  se cumple el criterio de decisión y por lo tanto se rechaza H0

####**TH para la media con varianza poblacional desconocida**



Sea $X \sim N(\mu ; \sigma )$

${t_{obs} =} {{\bar{X} - \mu } \over {S \over \sqrt n}}$ $\sim t_{n-1}$

Hipótesis y Criterio de decisión para rechazar H0 (C.D)

Prueba unilateral a derecha
\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: \mu \leq a     \\
		 H1: \mu > a
	       \end{array}
\thinspace C.D: si \thinspace t_{obs} > t_{n-1;1-\alpha} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

Prueba unilateral a izquierda

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: \mu \geq a     \\
		 H1: \mu < a
	       \end{array}
\thinspace C.D: si \thinspace t_{obs} < t_{n-1;\alpha} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

Prueba bilateral

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: \mu = a     \\
		 H1: \mu \neq a
	       \end{array}
\thinspace C.D: si \thinspace |t_{obs}| > t_{n-1;1-{\alpha \over 2}} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

Ejemplo: Sea X con distribución normal con media 31. Se toma una muestra de tamaño 20 y se obtuvo una media de 35 y un desvío de 2.5. Se quiere decidir si la media poblacional es mayor a 31 con α = 0.05

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: \mu \leq 31     \\
		 H1: \mu > 31
	       \end{array}
\thinspace C.D: si \thinspace t_{obs} > t_{n-1;1-\alpha} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

In [ ]:
media_muestral=35
mu= 31
S= 2.5
n= 20
alfa= 0.05
simbolo_H1='>'#las opciones de símbolo son '<' '>' ó '' para distinto

tobs=round((media_muestral-mu) / (S/np.sqrt(n)),4)
if simbolo_H1=='>':
  talfa  = round(ss.t.ppf(1-alfa, n-1),4)
  if tobs > talfa:
    respuesta='Como se cumple C.D ya que ' + 'tobs='+str(tobs)+' es mayor que '+ 't_1-alfa;n-1='+str(talfa)+' entonces se rechaza H0'
  else:
    respuesta='Como no se cumple C.D ya que ' + 'tobs='+str(tobs)+' es menor que '+ 't_1-alfa;n-1='+str(talfa)+' entonces no se rechaza H0'

  p_valor=round(1-ss.t.cdf(tobs, n-1),8)
  if p_valor < alfa:
    respuesta1='Como se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es menor que '+ 'alfa='+ str(alfa)+' entonces se rechaza H0'
  else:
    respuesta1='Como no se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es mayor que '+ 'alfa='+ str(alfa)+' entonces no se rechaza H0'

elif simbolo_H1=='<':
  talfa  = round(ss.t.ppf(alfa, n-1),4)
  if tobs < talfa:
    respuesta='Como se cumple C.D ya que ' + 'tobs='+str(tobs)+' es menor que '+ 'talfa;n-1='+str(talfa)+' entonces se rechaza H0'
  else:
    respuesta='Como no se cumple C.D ya que ' + 'tobs='+str(tobs)+' es mayor que '+ 'talfa;n-1='+str(talfa)+' entonces no se rechaza H0'

  p_valor=round(ss.t.cdf(tobs, n-1),8)
  if p_valor < alfa:
    respuesta1='Como se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es menor que '+ 'alfa='+ str(alfa)+' entonces se rechaza H0'
  else:
    respuesta1='Como no se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es mayor que '+ 'alfa='+ str(alfa)+' entonces no se rechaza H0'

else:
  talfa  = round(ss.t.ppf(1-alfa/2, n-1),4)
  if np.abs(tobs) > talfa:
    respuesta='Como se cumple C.D ya que ' + '|tobs|='+'|'+str(tobs)+'|'+' es mayor que '+ 't_1-alfa/2;n-1='+str(talfa)+' entonces se rechaza H0'
  else:
    respuesta='Como no se cumple C.D ya que ' + '|tobs|='+'|'+str(tobs)+'|'+' es menor que '+ 't_1-alfa/2;n-1='+str(talfa)+' entonces no se rechaza H0'

  p_valor=2*round(1-ss.t.cdf(abs(tobs), n-1),8)
  if p_valor < alfa:
    respuesta1='Como se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es menor que '+ 'alfa='+ str(alfa)+' entonces se rechaza H0'
  else:
    respuesta1='Como no se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es mayor que '+ 'alfa='+ str(alfa)+' entonces no se rechaza H0'

respuesta, respuesta1

('Como se cumple C.D ya que tobs=7.1554 es mayor que t_1-alfa;n-1=1.7291 entonces se rechaza H0',
 'Como se cumple C.D ya que P-valor=4.2e-07 es menor que alfa=0.05 entonces se rechaza H0')

Se verifica este resultado del siguiente modo:

Dados $\bar{X}=35$ $S=2.5$ $N=20$ y $\alpha=5\%$

Luego:

$t_{1-{\alpha};n-1}=t_{1-{0.05};20-1}=t_{0.95;19} = 1.73$

${t_{obs} =} {{\bar{X} - \mu } \over {S \over \sqrt n}}={{35 - 31} \over {2.5 \over \sqrt 20}}={7.1554}$

Como $|{t_{obs}| =}{7.1554}$ es menor que $t_{0.95:19}= 1.73$ se cumple el criterio de decisión y por lo tanto se rechaza H0

Como $p_{valor}= p(t > 7.1554)=1-p(t < 7.1554)=1-1=0$ es menor que $\alpha = 0.05$  se cumple el criterio de decisión y por lo tanto se rechaza H0

####**TH para la varianza**



Sea $X \sim N(\mu ; \sigma )$

${\chi_{obs} =} {{(n-1)*S^2 } \over {\sigma^2}}$ $\sim \chi^2_{n-1}$

Hipótesis y Criterio de decisión para rechazar H0 (C.D)

Prueba unilateral a derecha

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: \sigma^2 \leq a     \\
		 H1: \sigma^2 > a        
	       \end{array}
 \thinspace C.D: si \thinspace \chi_{obs} > \chi^2_{n-1;1-\alpha} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

Prueba unilateral a izquierda

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: \sigma^2 \geq a     \\
		 H1: \sigma^2 < a        
	       \end{array}
\thinspace C.D: si \thinspace \chi_{obs} < \chi^2_{n-1;\alpha} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

Prueba bilateral

\begin{equation}
\left\{
	       \begin{array}{ll}
		 H0: \sigma^2 = a     \\
		 H1: \sigma^2 \neq a         
	       \end{array}
\thinspace C.D: si \thinspace |\chi_{obs}| > \chi^2_{n-1;{1-{\alpha \over 2}}} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

EJEMPLO: Sea X con distribución normal y varianza 80. Se toma una muestra de tamaño 30 y se obtiene una varianza de 90. Con un alfa de 0.05, ¿concluiría que la varianza poblacional difiere de 80?

\begin{equation}
\left\{
	       \begin{array}{ll}
		 H0: \sigma^2 = 80     \\
		 H1: \sigma^2 \neq 80         
	       \end{array}
\thinspace C.D: si \thinspace |\chi_{obs}| > \chi^2_{n-1;{1-{\alpha \over 2}}} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

In [ ]:
var_muestral=90
var_pob= 80
n= 30
alfa= 0.05
simbolo_H1=''   #las opciones de símbolo son '<' '>' ó '' para distinto

chiobs=round((n-1)*var_muestral / var_pob,4)
if simbolo_H1=='>':
  chialfa  = round(ss.chi2.ppf(1-alfa, n-1),4)
  if chiobs > chialfa:
    respuesta='Como se cumple C.D ya que ' + 'chiobs='+str(chiobs)+' es mayor que '+ 'Chi_1-alfa;n-1='+str(chialfa)+' entonces se rechaza H0'
  else:
    respuesta='Como no se cumple C.D ya que ' + 'chiobs='+str(chiobs)+' es menor que '+ 'Chi_1-alfa;n-1='+str(chialfa)+' entonces no se rechaza H0'

  p_valor=round(1-ss.chi2.cdf(chiobs, n-1),8)
  if p_valor < alfa:
    respuesta1='Como se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es menor que '+ 'alfa='+ str(alfa)+' entonces se rechaza H0'
  else:
    respuesta1='Como no se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es mayor que '+ 'alfa='+ str(alfa)+' entonces no se rechaza H0'

elif simbolo_H1=='<':
  chialfa  = round(ss.chi2.ppf(alfa, n-1),4)
  if chiobs < chialfa:
    respuesta='Como se cumple C.D ya que ' + 'chiobs='+str(chiobs)+' es menor que '+ 'chialfa;n-1='+str(chialfa)+' entonces se rechaza H0'
  else:
    respuesta='Como no se cumple C.D ya que ' + 'chiobs='+str(chiobs)+' es mayor que '+ 'chialfa;n-1='+str(chialfa)+' entonces no se rechaza H0'

  p_valor=round(ss.chi2.cdf(chiobs, n-1),8)
  if p_valor < alfa:
    respuesta1='Como se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es menor que '+ 'alfa='+ str(alfa)+' entonces se rechaza H0'
  else:
    respuesta1='Como no se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es mayor que '+ 'alfa='+ str(alfa)+' entonces no se rechaza H0'

else:
  chialfa  = round(ss.chi2.ppf(1-alfa/2, n-1),4)
  if np.abs(chiobs) > chialfa:
    respuesta='Como se cumple C.D ya que ' + '|chiobs|='+'|'+str(chiobs)+'|'+' es mayor que '+ 'Chi_1-alfa/2;n-1='+str(chialfa)+' entonces se rechaza H0'
  else:
    respuesta='Como no se cumple C.D ya que ' + '|chiobs|='+'|'+str(chiobs)+'|'+' es menor que '+ 'Chi_1-alfa/2;n-1='+str(chialfa)+' entonces no se rechaza H0'

  p_valor=2*np.min([round(1-ss.chi2.cdf(chiobs, n-1),8), round(ss.chi2.cdf(chiobs, n-1),8)])
  if p_valor < alfa:
    respuesta1='Como se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es menor que '+ 'alfa='+ str(alfa)+' entonces se rechaza H0'
  else:
    respuesta1='Como no se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es mayor que '+ 'alfa='+ str(alfa)+' entonces no se rechaza H0'

respuesta, respuesta1

('Como no se cumple C.D ya que |chiobs|=|32.625| es menor que Chi_1-alfa/2;n-1=45.7223 entonces no se rechaza H0',
 'Como no se cumple C.D ya que P-valor=0.5862414 es mayor que alfa=0.05 entonces no se rechaza H0')

Se verifica este resultado del siguiente modo:

Dados $S^2=90$ $\sigma^2=80$ $N=30$ y $\alpha=5\%$

Luego:

$\chi_{1-{\alpha\over2};n-1}=\chi_{1-{0.05\over2};25-1}=\chi_{0.975;29} = 45.7223$

${\chi_{obs} =} {{(n-1)*S^2 } \over {\sigma^2}}={{(30-1)*90} \over {80}}={32.625}$

Como ${\chi_{obs} =}{32.625}$ es menor que $\chi_{1-{\alpha\over2};n-1}= 45.7223$ no se cumple el criterio de decisión y por lo tanto no se rechaza H0

Como $p_{valor}= 2*min[p(\chi > 32.625), p(\chi < 32.625)]=2*[0.29312, 0.70688]=2*0.29312=0.58624$ \
es mayor que $\alpha = 0.05$ no se cumple el criterio de decisión y por lo tanto no se rechaza H0

####**TH para la proporción**



Sea $X \sim Bi(n ; p)$

${Z_{obs} =} {{\bar{P} - P } \over {\sqrt {P*(1-P) \over n}}}$ $\sim N(0 ; 1)$

Hipótesis y Criterio de decisión para rechazar H0 (C.D)

Prueba unilateral a derecha

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: p \leq a     \\
		 H1: p > a         
	       \end{array}
\thinspace C.D: si \thinspace Z_{obs} > Z_{1-\alpha} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

Prueba unilateral a izquierda

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: p \geq a     \\
		 H1: p < a         
	       \end{array}
\thinspace C.D: si \thinspace Z_{obs} < Z_{\alpha} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

Prueba bilateral

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: p = a     \\
		 H1: p \neq a         
	       \end{array}
\thinspace C.D: si \thinspace |Z_{obs}| > Z_{1-{\alpha \over 2}} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

Ejemplo: Sea X con distribución binomial. La proporción poblacional es de 0.05. Se toma una muestra de 1125 y se obtuvo una proporción de 0.0444. Se quiere decidir si la proporción poblacional es menor a 0.05 con α = 0.01

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: p \geq 0.05     \\
		 H1: p < 0.05         
	       \end{array}
\thinspace C.D: si \thinspace Z_{obs} < Z_{\alpha} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

In [ ]:
p_muestral= 0.0444
p= 0.05
n= 1125
alfa= 0.01
simbolo_H1= '<' #las opciones de símbolo son '<' '>' ó '' para distinto

Zobs=round((p_muestral-p) / (np.sqrt((p*(1-p))/n)),4)
if simbolo_H1=='>':
  Zalfa  = round(ss.norm.ppf(1-alfa, 0,1),4)
  if Zobs > Zalfa:
    respuesta='Como se cumple C.D ya que ' + 'Zobs='+str(Zobs)+' es mayor que '+ 'Z_1-alfa='+str(Zalfa)+' entonces se rechaza H0'
  else:
    respuesta='Como no se cumple C.D ya que ' + 'Zobs='+str(Zobs)+' es menor que '+ 'Z_1-alfa='+str(Zalfa)+' entonces no se rechaza H0'

  p_valor=1-ss.norm.cdf(Zobs, 0,1)
  if p_valor < alfa:
    respuesta1='Como se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es menor que '+ 'alfa='+ str(alfa)+' entonces se rechaza H0'
  else:
    respuesta1='Como no se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es mayor que '+ 'alfa='+ str(alfa)+' entonces no se rechaza H0'

elif simbolo_H1=='<':
  Zalfa  = round(ss.norm.ppf(alfa, 0,1),4)
  if Zobs < Zalfa:
    respuesta='Como se cumple C.D ya que ' + 'Zobs='+str(Zobs)+' es menor que '+ 'Zalfa='+str(Zalfa)+' entonces se rechaza H0'
  else:
    respuesta='Como no se cumple C.D ya que ' + 'Zobs='+str(Zobs)+' es mayor que '+ 'Zalfa='+str(Zalfa)+' entonces no se rechaza H0'

  p_valor=ss.norm.cdf(Zobs, 0,1)
  if p_valor < alfa:
    respuesta1='Como se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es menor que '+ 'alfa='+ str(alfa)+' entonces se rechaza H0'
  else:
    respuesta1='Como no se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es mayor que '+ 'alfa='+ str(alfa)+' entonces no se rechaza H0'

else:
  Zalfa  = round(ss.norm.ppf(1-alfa/2, 0,1),4)
  if np.abs(Zobs) > Zalfa:
    respuesta='Como se cumple C.D ya que ' + '|Zobs|='+'|'+str(Zobs)+'|'+' es mayor que '+ 'Z_1-alfa/2='+str(Zalfa)+' entonces se rechaza H0'
  else:
    respuesta='Como no se cumple C.D ya que ' + '|Zobs|='+'|'+str(Zobs)+'|'+' es menor que '+ 'Z_1-alfa/2='+str(Zalfa)+' entonces no se rechaza H0'

  p_valor=2*(1-ss.norm.cdf(abs(Zobs), 0,1))
  if p_valor < alfa:
    respuesta1='Como se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es menor que '+ 'alfa='+ str(alfa)+' entonces se rechaza H0'
  else:
    respuesta1='Como no se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es mayor que '+ 'alfa='+ str(alfa)+' entonces no se rechaza H0'

respuesta, respuesta1

('Como no se cumple C.D ya que Zobs=-0.8618 es mayor que Zalfa=-2.3263 entonces no se rechaza H0',
 'Como no se cumple C.D ya que P-valor=0.19439879246780117 es mayor que alfa=0.01 entonces no se rechaza H0')

Se verifica este resultado del siguiente modo:

Dados $\bar{P}=0.0444$ $N=1125$ y $\alpha=1\%$

Luego:

$Z_{\alpha}=Z_{0.01}= - 2.33$


${Z_{obs} =} {{\bar{P} - P } \over {\sqrt {P*(1-P) \over n}}}={{0.0444 - 0.05} \over {\sqrt {0.05*(1-0.05) \over 1125}}}={-0.8618}$

Como ${Z_{obs} =}{-0.8618}$ es mayor que $Z_{0.05}= - 2.33$ no se cumple el criterio de decisión y por lo tanto no se rechaza H0

Como $p_{valor}= p(Z < -0.8618)=1-p(Z < 0.8618)=1-0.8056=0.19440$ es mayor que $\alpha = 0.01$  no se cumple el criterio de decisión y por lo tanto no se rechaza H0

##**Para dos poblaciones**

####**TH para la diferencia de medias con varianzas poblacionales conocidas**



Sea $X_{1} \sim N(\mu_{1} ; \sigma_{1} )$ y $X_{2} \sim N(\mu_{2} ; \sigma_{2} )$

${Z_{obs} =} {({\bar{X_{1}}-\bar{X_{2}}) - (\mu_{1} - \mu_{2}) } \over {\sqrt {{\sigma^2_{1} \over  n_{1}} + {\sigma^2_{2} \over  n_{2}}} }}$ $\sim N(0 ; 1)$

Hipótesis y Criterio de decisión para rechazar H0 (C.D)

Prueba unilateral a derecha

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: \mu_1 - \mu_2 \leq a     \\
		 H1: \mu_1 - \mu_2 > a         
	       \end{array}
\thinspace C.D: si \thinspace Z_{obs} > Z_{1-\alpha} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

Prueba unilateral a izquierda

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: \mu_1 - \mu_2 \geq a     \\
		 H1: \mu_1 - \mu_2 < a         
	       \end{array}
\thinspace C.D: si \thinspace Z_{obs} < Z_{\alpha} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

Prueba bilateral

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: \mu_1 - \mu_2 = a     \\
		 H1: \mu_1 - \mu_2 \neq a         
	       \end{array}
\thinspace C.D: si \thinspace |Z_{obs}| > Z_{1-{\alpha \over 2}} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

Ejemplo: Sea $X_1$ con distribución normal y varianza 0.64. Sea $X_2$ con distribución normal y varianza 0.49. Se toma una muestra para 1 de tamaño 50 y se obtiene una media de 45. Se toma una muestra para 2 de tamaño 40 y se obtiene una media de 43.2. Se quiere decidir si la media poblacional de 1 es 1.5 más grande que la media poblacional de 2 con α = 0.05

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: \mu_1 - \mu_2 \leq 1.5     \\
		 H1: \mu_1 - \mu_2 > 1.5         
	       \end{array}
\thinspace C.D: si \thinspace Z_{obs} > Z_{1-\alpha} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

In [ ]:
MM_1= 45
MM_2= 43.2
DIF= 1.5
sigma2_1=0.64
sigma2_2=0.49
n_1= 50
n_2= 40
alfa= 0.05
simbolo_H1='>'


Zobs=round((MM_1-MM_2)-DIF / (np.sqrt((sigma2_1/n_1)+(sigma2_2/n_2))),4)
if simbolo_H1=='>':
  Zalfa  = round(ss.norm.ppf(1-alfa, 0,1),4)
  if Zobs > Zalfa:
    respuesta='Como se cumple C.D ya que ' + 'Zobs='+str(Zobs)+' es mayor que '+ 'Z_1-alfa='+str(Zalfa)+' entonces se rechaza H0'
  else:
    respuesta='Como no se cumple C.D ya que ' + 'Zobs='+str(Zobs)+' es menor que '+ 'Z_1-alfa='+str(Zalfa)+' entonces no se rechaza H0'

  p_valor=1-ss.norm.cdf(Zobs, 0,1)
  if p_valor < alfa:
    respuesta1='Como se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es menor que '+ 'alfa='+ str(alfa)+' entonces se rechaza H0'
  else:
    respuesta1='Como no se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es mayor que '+ 'alfa='+ str(alfa)+' entonces no se rechaza H0'

elif simbolo_H1=='<':
  Zalfa  = round(ss.norm.ppf(alfa, 0,1),4)
  if Zobs < Zalfa:
    respuesta='Como se cumple C.D ya que ' + 'Zobs='+str(Zobs)+' es menor que '+ 'Zalfa='+str(Zalfa)+' entonces se rechaza H0'
  else:
    respuesta='Como no se cumple C.D ya que ' + 'Zobs='+str(Zobs)+' es mayor que '+ 'Zalfa='+str(Zalfa)+' entonces no se rechaza H0'

  p_valor=ss.norm.cdf(Zobs, 0,1)
  if p_valor < alfa:
    respuesta1='Como se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es menor que '+ 'alfa='+ str(alfa)+' entonces se rechaza H0'
  else:
    respuesta1='Como no se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es mayor que '+ 'alfa='+ str(alfa)+' entonces no se rechaza H0'

else:
  Zalfa  = round(ss.norm.ppf(1-alfa/2, 0,1),4)
  if np.abs(Zobs) > Zalfa:
    respuesta='Como se cumple C.D ya que ' + '|Zobs|='+'|'+str(Zobs)+'|'+' es mayor que '+ 'Z_1-alfa/2='+str(Zalfa)+' entonces se rechaza H0'
  else:
    respuesta='Como no se cumple C.D ya que ' + '|Zobs|='+'|'+str(Zobs)+'|'+' es menor que '+ 'Z_1-alfa/2='+str(Zalfa)+' entonces no se rechaza H0'

  p_valor=2*(1-ss.norm.cdf(abs(Zobs), 0,1))
  if p_valor < alfa:
    respuesta1='Como se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es menor que '+ 'alfa='+ str(alfa)+' entonces se rechaza H0'
  else:
    respuesta1='Como no se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es mayor que '+ 'alfa='+ str(alfa)+' entonces no se rechaza H0'

respuesta, respuesta1

('Como no se cumple C.D ya que Zobs=-7.6774 es menor que Z_1-alfa=1.6449 entonces no se rechaza H0',
 'Como no se cumple C.D ya que P-valor=0.9999999999999919 es mayor que alfa=0.05 entonces no se rechaza H0')

Se verifica este resultado del siguiente modo:

Dados $\bar{X_1}=45$ $\bar{X_2}=43.2$ $\sigma^2_1=0.64$ $\sigma^2_2=0.49$ $N_1=50$ $N_2=40$ y $\alpha=5\%$

Luego:

$Z_{1-\alpha}=Z_{1-0.05}=Z_{0.95} = 1.65$


${Z_{obs} =} {({\bar{X_{1}}-\bar{X_{2}}) - (\mu_{1} - \mu_{2}) } \over {\sqrt {{\sigma^2_{1} \over  n_{1}} + {\sigma^2_{2} \over  n_{2}}} }} = {(45-43.2) - 1.5  \over {\sqrt {{0.64 \over  50} + {0.49 \over  40}} }}= -7.6774$

Como ${Z_{obs} =}{-7.6774}$ es mayor que $Z_{0.95}= 1.65$ no se cumple el criterio de decisión y por lo tanto no se rechaza H0

Como $p_{valor}= p(Z > -7.6774)=1-p(Z > -7.6774)=1-[1-p(Z < 7.6774)]=p(Z < 7.6774)=0.9999999$ es mayor que $\alpha = 0.05$  no se cumple el criterio de decisión y por lo tanto no se rechaza H0

####**TH para la diferencia de medias con varianzas poblacionales desconocidas**



Sea $X_{1} \sim N(\mu_{1} ; \sigma_{1} )$ y $X_{2} \sim N(\mu_{2} ; \sigma_{2} )$ con $\sigma^2_{1}$ y $\sigma^2_{2}$ desconocidas y $\sigma^2_{1}=\sigma^2_{2} $

${t_{obs} =} {({\bar{X_{1}}-\bar{X_{2}}) - (\mu_{1} - \mu_{2}) } \over {S_{p}\sqrt {{1 \over  n_{1}} + {1 \over  n_{2}}} }}$ $\sim t_{n_{1}+n_{2}-2}$ con ${S^2_p =} {(n_1-1)*S^2_1+(n_2-1)*S^2_2\over n_1+n_2-2}$

Hipótesis y Criterio de decisión para rechazar H0 (C.D)

Prueba unilateral a derecha

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: \mu_1 - \mu_2 \leq a     \\
		 H1: \mu_1 - \mu_2 > a         
	       \end{array}
\thinspace C.D: si \thinspace t_{obs} > t_{n_{1}+n_{2}-2;1-\alpha} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

Prueba unilateral a izquierda

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: \mu_1 - \mu_2 \geq a     \\
		 H1: \mu_1 - \mu_2 < a        
	       \end{array}
 \thinspace C.D: si \thinspace t_{obs} < t_{n_{1}+n_{2}-2;\alpha} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

Prueba bilateral

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: \mu_1 - \mu_2 = a     \\
		 H1: \mu_1 - \mu_2 \neq a         
	       \end{array}
\thinspace C.D: si \thinspace |t_{obs}| > t_{n_{1}+n_{2}-2;1-{\alpha \over 2}} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

Ejemplo: Sea $X_1$ y $X_2$ con distribución normal. Se toma una muestra para 1 de tamaño 50 y se obtiene una media de 35735.45 y una varianza de 25318.41. Se toma una muestra para 2 de tamaño 50 y se obtiene una media de 35732.03 y una varianza de 25337.03 Se quiere decidir si la media poblacional de 1 difiere de la media poblacional de 2 con α = 0.05

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: \mu_1 - \mu_2 = 0     \\
		 H1: \mu_1 - \mu_2 \neq 0         
	       \end{array}
\thinspace C.D: si \thinspace |t_{obs}| > t_{n_{1}+n_{2}-2;1-{\alpha \over 2}} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

In [ ]:
MM_1=35735.45
MM_2= 35732.03
DIF= 0
varM_1= 25318.41
varM_2= 25337.03
n_1= 50
n_2= 50
alfa= 0.05
simbolo_H1=''

Sp=np.sqrt(((n_1-1)*(varM_1) + (n_2-1)*(varM_2))/(n_1+n_2-2))
tobs=round(((MM_1-MM_2)-DIF) / (Sp*(np.sqrt((1/n_1)+(1/n_2)))),4)
if simbolo_H1=='>':
  talfa  = round(ss.t.ppf(1-alfa, n_1+n_2-2),4)
  if tobs > talfa:
    respuesta='Como se cumple C.D ya que ' + 'tobs='+str(tobs)+' es mayor que '+ 't_1-alfa;n_1+n_2-2='+str(talfa)+' entonces se rechaza H0'
  else:
    respuesta='Como no se cumple C.D ya que ' + 'tobs='+str(tobs)+' es menor que '+ 't_1-alfa;n_1+n_2-2='+str(talfa)+' entonces no se rechaza H0'

  p_valor=round(1-ss.t.cdf(tobs, n_1+n_2-2),8)
  if p_valor < alfa:
    respuesta1='Como se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es menor que '+ 'alfa='+ str(alfa)+' entonces se rechaza H0'
  else:
    respuesta1='Como no se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es mayor que '+ 'alfa='+ str(alfa)+' entonces no se rechaza H0'

elif simbolo_H1=='<':
  talfa  = round(ss.t.ppf(alfa, n_1+n_2-2),4)
  if tobs < talfa:
    respuesta='Como se cumple C.D ya que ' + 'tobs='+str(tobs)+' es menor que '+ 'talfa;n_1+n_2-2='+str(talfa)+' entonces se rechaza H0'
  else:
    respuesta='Como no se cumple C.D ya que ' + 'tobs='+str(tobs)+' es mayor que '+ 'talfa;n_1+n_2-2='+str(talfa)+' entonces no se rechaza H0'

  p_valor=round(ss.t.cdf(tobs, n_1+n_2-2),8)
  if p_valor < alfa:
    respuesta1='Como se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es menor que '+ 'alfa='+ str(alfa)+' entonces se rechaza H0'
  else:
    respuesta1='Como no se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es mayor que '+ 'alfa='+ str(alfa)+' entonces no se rechaza H0'

else:
  talfa  = round(ss.t.ppf(1-alfa/2, n_1+n_2-2),4)
  if np.abs(tobs) > talfa:
    respuesta='Como se cumple C.D ya que ' + '|tobs|='+'|'+str(tobs)+'|'+' es mayor que '+ 't_1-alfa/2;n_1+n_2-2='+str(talfa)+' entonces se rechaza H0'
  else:
    respuesta='Como no se cumple C.D ya que ' + '|tobs|='+'|'+str(tobs)+'|'+' es menor que '+ 't_1-alfa/2;n_1+n_2-2='+str(talfa)+' entonces no se rechaza H0'

  p_valor=2*round(1-ss.t.cdf(abs(tobs), n_1+n_2-2),8)
  if p_valor < alfa:
    respuesta1='Como se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es menor que '+ 'alfa='+ str(alfa)+' entonces se rechaza H0'
  else:
    respuesta1='Como no se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es mayor que '+ 'alfa='+ str(alfa)+' entonces no se rechaza H0'

respuesta, respuesta1

('Como no se cumple C.D ya que |tobs|=|0.1074| es menor que t_1-alfa/2;n_1+n_2-2=1.9845 entonces no se rechaza H0',
 'Como no se cumple C.D ya que P-valor=0.91469122 es mayor que alfa=0.05 entonces no se rechaza H0')

Se verifica este resultado del siguiente modo:

Dados $\bar{X_1}=35735.45$ $\bar{X_2}=35732.03$ $S^2_1=25318.41$ $S^2_2=25337.03$ $N_1=50$ $N_2=50$ y $\alpha=5\%$

Luego:

$t_{n_{1}+n_{2}-2;1-{\alpha\over2}}=t_{50+50-2;1-{0.05\over2}}=t_{98;0.975} = 1.9845$

${S^2_p =} {(n_1-1)*S^2_1+(n_2-1)*S^2_2\over n_1+n_2-2}= {(50-1)*25318.41+(50-1)*25337.03\over 50+50-2}=25327.72$

${S_p =}{\sqrt{S^2_p}}={\sqrt{25327.72}}=159.1468504$

${t_{obs} =} {({\bar{X_{1}}-\bar{X_{2}}) - (\mu_{1} - \mu_{2}) } \over {S_{p}\sqrt {{1 \over  n_{1}} + {1 \over  n_{2}}}}}={(35735.45-35732.03) - 0 \over {159.1468504\sqrt {{1 \over  50} + {1 \over  50}}}}=0.1074$  

Como ${t_{obs} =}{|0.1074|}$ es menor que $t_{98;0.975} = 1.9845$ no se cumple el criterio de decisión y por lo tanto no se rechaza H0

Como $p_{valor}= p(t > 0.1074)=1-p(t < 0.1074)= 1 - 0.54265= 0.45735$ es mayor que $\alpha = 0.05$  no se cumple el criterio de decisión y por lo tanto no se rechaza H0

####**TH para igualdad de varianzas**



Sea $X_{1} \sim N(\mu_{1} ; \sigma_{1} )$ y $X_{2} \sim N(\mu_{2} ; \sigma_{2} )$

${F_{obs} =}  {S^2_{1} \over  S^2_{2}}$ $\sim F_{n_{1}-1; n_{2}-1}$

Hipótesis y Criterio de decisión para rechazar H0 (C.D)

Prueba unilateral a derecha

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: \sigma^2_{1} \leq \sigma^2_{2}     \\
		 H1: \sigma^2_{1} > \sigma^2_{2}         
	       \end{array}
\thinspace C.D: si \thinspace F_{obs} > F_{n_1-1;n_2-1;1-\alpha} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

Prueba unilateral a izquierda

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: \sigma^2_{1} \geq \sigma^2_{2}    \\
		 H1: \sigma^2_{1} < \sigma^2_{2}        
	       \end{array}
\thinspace C.D: si \thinspace F_{obs} < F_{n_1-1;n_2-1;\alpha} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

Prueba bilateral

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: \sigma^2_{1} = \sigma^2_{2}     \\
		 H1: \sigma^2_{1} \neq \sigma^2_{2}         
	       \end{array}
\thinspace C.D: si \thinspace |F_{obs}| > F_{n_1-1;n_2-1;{1-{\alpha \over 2}}} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

Se crea una función para calcular el test

Ejemplo: Sea $X_1$ y $X_2$ con distribución normal y suponga que $\sigma^2_{1}=\sigma^2_{2} $. Se toma una muestra para 1 de tamaño 10 y se obtiene una media de 4.2 y una varianza de 0.3364. Se toma una muestra para 2 de tamaño 10 y se obtiene una media de 5.25 y una varianza de 0.3969. Se quiere validar el supuesto de que las varianzas poblacionales son iguales con α = 0.05.

\begin{equation}
\left\{
	       \begin{array}{cc}
		 H0: \sigma^2_{1} = \sigma^2_{2}     \\
		 H1: \sigma^2_{1} \neq \sigma^2_{2}         
	       \end{array}
\thinspace C.D: si \thinspace |F_{obs}| > F_{n_1-1;n_2-1;{1-{\alpha \over 2}}} \thinspace ó \thinspace p-valor < \alpha \thinspace entonces \thinspace rechazo \thinspace H0
	     \right.
\end{equation}

In [ ]:
var_M1=0.3364
var_M2= 0.3969
n_1= 10
n_2= 10
alfa= 0.05
simbolo_H1=''

Fobs=round(var_M1 / var_M2,4)
if simbolo_H1=='>':
  falfa  = round(ss.f.ppf(1-alfa, n_1-1,n_2-1),4)
  if Fobs > falfa:
    respuesta='Como se cumple C.D ya que ' + 'Fobs='+str(Fobs)+' es mayor que '+ 'F_1-alfa;n1-1;n2-1='+str(falfa)+' entonces se rechaza H0'
  else:
    respuesta='Como no se cumple C.D ya que ' + 'Fobs='+str(Fobs)+' es menor que '+ 'F_1-alfa;n1-1;n2-1='+str(falfa)+' entonces no se rechaza H0'

  p_valor=round(1-ss.f.cdf(Fobs, n_1-1,n_2-1),8)
  if p_valor < alfa:
    respuesta1='Como se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es menor que '+ 'alfa='+ str(alfa)+' entonces se rechaza H0'
  else:
    respuesta1='Como no se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es mayor que '+ 'alfa='+ str(alfa)+' entonces no se rechaza H0'

elif simbolo_H1=='<':
  falfa  = round(ss.f.ppf(alfa, n_1-1,n_2-1),4)
  if Fobs < falfa:
    respuesta='Como se cumple C.D ya que ' + 'Fobs='+str(Fobs)+' es menor que '+ 'F_alfa;n1-1;n2-1='+str(falfa)+' entonces se rechaza H0'
  else:
    respuesta='Como no se cumple C.D ya que ' + 'Fobs='+str(Fobs)+' es mayor que '+ 'F_alfa;n1-1;n2-1='+str(falfa)+' entonces no se rechaza H0'

  p_valor=round(ss.f.cdf(Fobs, n_1-1,n_2-1),8)
  if p_valor < alfa:
    respuesta1='Como se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es menor que '+ 'alfa='+ str(alfa)+' entonces se rechaza H0'
  else:
    respuesta1='Como no se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es mayor que '+ 'alfa='+ str(alfa)+' entonces no se rechaza H0'

else:
  falfa  = round(ss.f.ppf(1-alfa/2, n_1-1,n_2-1),4)
  if np.abs(Fobs) > falfa:
    respuesta='Como se cumple C.D ya que ' + '|Fobs|='+'|'+str(Fobs)+'|'+' es mayor que '+ 'F_1-alfa/2;n1-1;n2-1='+str(falfa)+' entonces se rechaza H0'
  else:
    respuesta='Como no se cumple C.D ya que ' + '|Fobs|='+'|'+str(Fobs)+'|'+' es menor que '+ 'F_1-alfa/2;n1-1;n2-1='+str(falfa)+' entonces no se rechaza H0'

  p_valor=2*np.min([round(1-ss.f.cdf(Fobs, n_1-1,n_2-1),8), round(ss.f.cdf(Fobs, n_1-1,n_2-1),8)])
  if p_valor < alfa:
    respuesta1='Como se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es menor que '+ 'alfa='+ str(alfa)+' entonces se rechaza H0'
  else:
    respuesta1='Como no se cumple C.D ya que ' + 'P-valor='+str(p_valor)+  ' es mayor que '+ 'alfa='+ str(alfa)+' entonces no se rechaza H0'

respuesta, respuesta1

('Como no se cumple C.D ya que |Fobs|=|0.8476| es menor que F_1-alfa/2;n1-1;n2-1=4.026 entonces no se rechaza H0',
 'Como no se cumple C.D ya que P-valor=0.80947338 es mayor que alfa=0.05 entonces no se rechaza H0')

Se verifica este resultado del siguiente modo:

Dados $S^2_1=0.3364$ $S^2_2=0.3969$ $N_1=10$ $N_2=10$ y $\alpha=5\%$

Luego:

$F_{n_{1}-1; n_{2}-1;1-{\alpha\over2}}=F_{10-1; 10-1;1-{0.05\over2}}=F_{9; 9;0.975}=4.026$

${F_{obs} =}  {S^2_{1} \over  S^2_{2}}={0.3364 \over  0.3969}=0.847568657$

Como ${F_{obs} =}{0.847568657}$ es menor que $F_{9;9;0.975} = 4.026$ no se cumple el criterio de decisión y por lo tanto no se rechaza H0

Como $p_{valor}= 2*min[p(F > 0.847568657);p(F < 0.847568657)]= 2*min[1-p(F < 0.847568657);p(F < 0.847568657)]=2*(0.40472)=0.80944$ es mayor que $\alpha = 0.05$ no se cumple el criterio de decisión y por lo tanto no se rechaza H0